File: 190_reverse_bits/reverse_bits.go
package main

import "fmt"

type Solution struct {
}

func (s *Solution) reverseBits(num uint32) uint32 {
	var res uint32 = 0
	for i := 0; i < 32; i++ {
		res = (res << 1) | (num & 1)
		num >>= 1
	}
	return res
}

func main() {
	s := Solution{}
	var n uint32 = 43261596
	fmt.Println(s.reverseBits(n))
}

File: 190_reverse_bits/reverse_bits.py
class Solution:
    def reverse_bits(self, n: int) -> int:
        res = 0
        for _ in range(32):
            res = (res << 1) | (n & 1)
            n >>= 1
        return res


if __name__ == "__main__":
    s = Solution()
    print(s.reverse_bits(43261596))
